test/slots/factories/SlotFactory.test.js

```diff
@@ -21,8 +25,22 @@ describe('the slot factory', () => {
       }).toThrow('Slot type not supported FALSEY')
     })
 
+    it('throws error when the type does not have a class defined for the provided dataType', () => {
+      const inSlotData = InSlotSeeder.generate()
+      const bogusDataType = 'trump'
+
+      expect(() => {
+        SlotFactory.new({ ...inSlotData, dataType: bogusDataType })
+      }).toThrow(`No ${bogusDataType} slot class found for ${inSlotData.type}`)
+    })
+
     describe('when making InSlots', () => {
-      Object.values(InSlot.DATA_TYPES).forEach(dataType => {
+      [
+        FloatInSlot.DATA_TYPE,
+        IntegerInSlot.DATA_TYPE,
+        BooleanInSlot.DATA_TYPE,
+        OneOfInSlot.DATA_TYPE,
+      ].forEach(dataType => {
```

Comment:
```suggestion
      const supportedDataTypes = SlotFactory.SUPPORTED_CLASSES.map(
        SlotClass => SlotClass.DATA_TYPE
      )
      
      supportedDataTypes.forEach(dataType => {
```

may make the test less brittle and/or auto extend it should someone add a new dataType